### PR TITLE
FLUT-6290 - [Others] RTL file changed and Localization example was added in the cartesian chart.

### PR DESCRIPTION
**Pull request by PavithraNandhiniRamachandran**
## Description ##

* Have changed some modifications in RTL sample for all charts(cartesian, circular, funnel and pyramid).
* Localization example was added in the cartesian chart.
* See also link was added the charts.

### CODE REVIEW

File: Flutter/cartesian-charts/right-to-left.md [hunk 1]
```diff
@@ -37,12 +37,13 @@ To change the rendering direction from right to left, you can wrap the [`SfCarte
 
 ### Changing the locale to RTL languages
 
-To change the chart rendering direction from right to left, you can change the [`locale`](https://api.flutter.dev/flutter/material/MaterialApp/locale.html) to any of the RTL languages such as Arabic, Persian, Hebrew, Pashto, and Urdu.
+To change the chart rendering direction from right to left, you can change the [`locale`](https://api.flutter.dev/flutter/material/MaterialApp/locale.html) to any of the RTL languages such as Arabic, Persian, Hebrew, Pashto, Urdu and so on.
```
Comment (dharanidharandharmasivam) [inline Flutter/cartesian-charts/right-to-left.md:40]: why you have added and so on.

**Review by dharanidharandharmasivam — CHANGES_REQUESTED** [review]

File: Flutter/cartesian-charts/right-to-left.md (hunk 1 above)
Comment (PavithraNandhiniRamachandran) [reply Flutter/cartesian-charts/right-to-left.md]: have resolved the concerns

File: Flutter/cartesian-charts/right-to-left.md [hunk 2]
```diff
@@ -9,9 +9,9 @@ documentation: ug
 
 # Right To Left (RTL) in Flutter Cartesian Chart (SfCartesianChart)
 
-Cartesian chart supports right to left rendering. But chart axis, series, and other chart elements rendering will be the same for both LTR and RTL. Only, the legend rendering will be changed.
+Cartesian chart supports right to left rendering. But chart axis, series, and other chart elements rendering will be the same for both LTR and RTL except trackball, tooltip and legend.
```
[the remark this answers is not in the document]
Comment (Marismathan) [reply Flutter/cartesian-charts/right-to-left.md]: Changed the content as mentioned above

File: Flutter/cartesian-charts/right-to-left.md [hunk 3]
```diff
@@ -23,94 +23,250 @@ To change the rendering direction from right to left, you can wrap the [`SfCarte
...
 
 {% endhighlight %}
 
-![series axis RTL](images/rtl-support/series_axis_rtl.jpg)
+![legend RTL](images/rtl-support/cartesian_legend_rtl.jpg)
 
-Also, if you want to change the tooltip’s content, to look like it is rendering from right to left, then you can set the [`format`](https://pub.dev/documentation/syncfusion_flutter_charts/latest/charts/TooltipBehavior/format.html) property in [`TooltipBehavior`](https://pub.dev/documentation/syncfusion_flutter_charts/latest/charts/TooltipBehavior-class.html) as `point.y : point.x`. By default, the tooltip format will be `point.x : point.y`.
+### Tooltip
+
+Right-to-left rendering is applicable for [`tooltip`](https://pub.dev/documentation/syncfusion_flutter_charts/latest/charts/TooltipBehavior-class.html) elements. The marker renders at first and following by that the tooltip renders on the right in LTR rendering. In RTL rendering, the tooltip content renders at first and following by that the marker on the right. By default, the tooltip content will be 'point.x : point.y'. In RTL rendering, the tooltip content will be 'point.y : point.x'.
```
[the remark this answers is not in the document]
Comment (Marismathan) [reply Flutter/cartesian-charts/right-to-left.md]: Changed the content as mentioned above

File: Flutter/cartesian-charts/right-to-left.md [hunk 4]
```diff
@@ -23,94 +23,250 @@ To change the rendering direction from right to left, you can wrap the [`SfCarte
...
+
+{% endhighlight %}
+
+![Tooltip RTL](images/rtl-support/cartesian_tooltip_rtl.jpg)
+
+### Trackball
+
+Right-to-left rendering is applicable for trackball tooltip elements.
+
+* By default, the trackball will render on the right side of the trackball line in LTR rendering if adequate space is provided.
```
[the remark this answers is not in the document]
Comment (Marismathan) [reply Flutter/cartesian-charts/right-to-left.md]: Changed the content as mentioned above

File: Flutter/cartesian-charts/Localization.md [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+platform: flutter
+control: Chart
+documentation: ug
+---
+
+# Localization in Flutter Cartesian Charts (SfCartesianChart)
+
+By default, the Cartesian chart widget supports US English localizations. You can change the other languages by specifying the [`MaterialApp`](https://api.flutter.dev/flutter/material/MaterialApp/MaterialApp.html) properties and adding the [`flutter_localizations`](https://pub.dev/packages/localization) package to your application.
+
+As of November 2020, [`flutter package`](https://flutter.dev/docs/development/accessibility-and-localization/internationalization) supports 77 languages.
```
[the remark this answers is not in the document]
Comment (Marismathan) [reply Flutter/cartesian-charts/Localization.md:14]: As per the discussion, I have removed that line as of now.

File: Flutter/cartesian-charts/Localization.md [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+
+To use [`flutter_localizations`](https://pub.dev/packages/localization), add the package as dependency to `pubspec.yaml` file.
+
+{% highlight dart %}
+
+    dependencies:
+    flutter_localizations:
```
[the remark this answers is not in the document]
Comment (Marismathan) [reply Flutter/cartesian-charts/Localization.md:21]: Aligned this code

File: Flutter/cartesian-charts/Localization.md [hunk 7]
```diff
@@ -0,0 +1,138 @@
...
+
+## Localize the custom text in chart
+
+Cartesian chart custom text can be localized using the [`syncfusion_localizations`](https://pub.dev/packages/syncfusion_localizations) package and specifying [`localizationsDelegates`](https://api.flutter.dev/flutter/material/MaterialApp/localizationsDelegates.html) in [`MaterialApp`](https://api.flutter.dev/flutter/material/MaterialApp/MaterialApp.html).
+
+To use [`syncfusion_localizations`](https://pub.dev/packages/syncfusion_localizations), add the package as dependency to `pubspec.yaml` file.
+
+{% highlight dart %}
+
+dependencies:
```
[the remark this answers is not in the document]
Comment (Marismathan) [reply Flutter/cartesian-charts/Localization.md:68]: Aligned the code

File: Flutter/cartesian-charts/Localization.md [hunk 8]
```diff
@@ -0,0 +1,138 @@
...
+    dependencies:
+    flutter_localizations:
+    sdk: flutter
+
+{% endhighlight %}
+
+Next, import the [`flutter_localizations`](https://pub.dev/packages/localization) library and specify [`localizationsDelegates`](https://api.flutter.dev/flutter/material/MaterialApp/localizationsDelegates.html) and [`supportedLocales`](https://api.flutter.dev/flutter/material/MaterialApp/supportedLocales.html) for [`MaterialApp`](https://api.flutter.dev/flutter/material/MaterialApp/MaterialApp.html).
+
+{% tabs %}
+{% highlight Dart %}
```
[the remark this answers is not in the document]
Comment (Marismathan) [reply Flutter/cartesian-charts/Localization.md:29]: Changed it as small

File: Flutter/cartesian-charts/Localization.md [hunk 9]
```diff
@@ -0,0 +1,138 @@
...
+    import 'package:flutter_localizations/flutter_localizations.dart';
+
+    @override
+    Widget build(BuildContext context) {
+      return MaterialApp(
+        localizationsDelegates: const [
+          GlobalMaterialLocalizations.delegate,
+          GlobalWidgetsLocalizations.delegate,
+        ],
+        supportedLocales: [
```
[the remark this answers is not in the document]
Comment (Marismathan) [reply Flutter/cartesian-charts/Localization.md:40]: Changed as mentioned

File: Flutter/cartesian-charts/Localization.md [hunk 10]
```diff
@@ -0,0 +1,138 @@
...
+
+Next, import the [`syncfusion_localizations`](https://pub.dev/packages/syncfusion_localizations) library.
+
+{% highlight dart %}
+
+    import 'package:syncfusion_localizations/syncfusion_localizations.dart';
+
+{% endhighlight %}
+
+Then, declare the [`SfGlobalLocalizations.delegate`](https://pub.dev/documentation/syncfusion_localizations/latest/syncfusion_localizations/SfGlobalLocalizations-class.html) in the [`localizationsDelegates`](https://api.flutter.dev/flutter/material/MaterialApp/localizationsDelegates.html), which is used to localize the custom string using in Cartesian and specify the [`supportedLocales`](https://api.flutter.dev/flutter/material/MaterialApp/supportedLocales.html) as well.
```
[the remark this answers is not in the document]
Comment (Marismathan) [reply Flutter/cartesian-charts/Localization.md:83]: Changed as mentioned

File: Flutter/cartesian-charts/right-to-left.md [hunk 11]
```diff
@@ -23,94 +23,249 @@ To change the rendering direction from right to left, you can wrap the [`SfCarte
...
+      ChartData(this.x, this.y);
+        final int x;
+        final int y;
+    }
 
 {% endhighlight %}
 
-![Tooltip RTL](images/rtl-support/cartesian_tooltip_rtl.png)
\ No newline at end of file
+![series axis RTL](images/rtl-support/series_rtl.jpg)
```
[the remark this answers is not in the document]
Comment (Marismathan) [reply Flutter/cartesian-charts/right-to-left.md:271]: Changed as mentioned

File: Flutter/cartesian-charts/Localization.md (hunk 10 above)
[the remark this answers is not in the document]
Comment (dharanidharandharmasivam) [reply Flutter/cartesian-charts/Localization.md]: Then, declare the [`SfGlobalLocalizations.delegate`](https://pub.dev/documentation/syncfusion_localizations/latest/syncfusion_localizations/SfGlobalLocalizations-class.html) in the [`localizationsDelegates`](https://api.flutter.dev/flutter/material/MaterialApp/localizationsDelegates.html), which is used to localize the custom string (series name displayed in legend) in the Cartesian chart and specify the [`supportedLocales`](https://api.flutter.dev/flutter/material/MaterialApp/supportedLocales.html) as well. Based on the current locale, the built-in text in the legend and tooltips are translated automatically.

File: Flutter/circular-charts/right-to-left.md [hunk 12]
```diff
@@ -41,56 +41,127 @@ To change the chart rendering direction from right to left, you can change the [
...
+        final int y;
+    }
+
+{% endhighlight %}
+
+![Legend RTL](images/rtl-support/circular_legend_rtl.jpg)
+
+### Tooltip
+
+Right-to-left rendering is applicable for [`tooltip`](https://pub.dev/documentation/syncfusion_flutter_charts/latest/charts/TooltipBehavior-class.html) elements. Here, the tooltip content renders at first and followed by that the marker on the right. By default, the tooltip content will be `point.x : point.y`, in RTL rendering, the tooltip content will be `point.y : point.x`. If you wish the format to be applied as it is despite RTL rendering in this case, you can make use of [`onTooltipRender`](https://pub.dev/documentation/syncfusion_flutter_charts/latest/charts/SfCircularChart/onTooltipRender.html) callback.
```
Comment (dharanidharandharmasivam) [inline Flutter/circular-charts/right-to-left.md:121]: Right-to-left rendering is applicable for [`tooltip`](https://pub.dev/documentation/syncfusion_flutter_charts/latest/charts/TooltipBehavior-class.html) elements. Here, the tooltip content renders at first and followed by that the marker on the right. By default, the tooltip content will be `point.x : point.y`, in RTL rendering the tooltip content will be `point.y : point.x`. If you wish the format to be applied as it is despite RTL rendering in this case, you can make use of [`onTooltipRender`](https://pub.dev/documentation/syncfusion_flutter_charts/latest/charts/SfCircularChart/onTooltipRender.html) callback.

File: Flutter/cartesian-charts/trackball-crosshair.md [hunk 13]
```diff
@@ -331,6 +331,82 @@ The ActivationMode enum contains the following values:
 
 ![Trackball tooltip overlap](images/trackball-crosshair/trackball_overlap.png)
 
+And moreover, the trackball tooltip will render on the right side of the trackball line if adequate space is available else will be moved to the left by default. But the trackball tooltip renders neatly across the trackball line within the available chart area when the chart size is smaller and the trackball tooltip cannot be placed on the sides of the trackball line.
```
Comment (dharanidharandharmasivam) [inline Flutter/cartesian-charts/trackball-crosshair.md:334]: And moreover, the trackball tooltip will render on the right side of the trackball line if adequate space is available else will be moved to the left by default.  The trackball tooltip is smartly positioned above the trackball line if there is no space on the right and left sides of the trackball line. However, the trackball tooltip size must be smaller than the chart's plot area size.

File: Flutter/cartesian-charts/Localization.md (hunk 10 above)
[the remark this answers is not in the document]
Comment (Marismathan) [reply Flutter/cartesian-charts/Localization.md]: Changed the above content as per we discussed in meeting

File: Flutter/circular-charts/right-to-left.md [hunk 14]
```diff
@@ -41,56 +41,127 @@ To change the chart rendering direction from right to left, you can change the [
...
 
-
 {% endhighlight %}
 
 ## RTL supported chart elements
 
-Right to left rendering is effective only for the legend in the chart. Legend items will be rendered from right to left direction.
+### Legend
+
+Right to left rendering is effective for the legend in the chart. Legend items will be rendered from right to left direction.
```
[the remark this answers is not in the document]
Comment (Marismathan) [reply Flutter/circular-charts/right-to-left.md]: Added the mentioned content

File: Flutter/circular-charts/right-to-left.md (hunk 12 above)
Comment (Marismathan) [reply Flutter/circular-charts/right-to-left.md]: Removed the comma in all RTL files

File: Flutter/cartesian-charts/trackball-crosshair.md (hunk 13 above)
Comment (Marismathan) [reply Flutter/cartesian-charts/trackball-crosshair.md]: Changed the content as mentioned

**Review by dharanidharandharmasivam — APPROVED** [review]
Hii @Marismathan ,
All the review changes were completed and approved.
Thanks,
Dharani.

**Pull request merged**